### DISCUSSION TITLE
Can be routing disable inside a specific SearchBox?

### DISCUSSION THREAD

**Discussion by Mat-moran** [algolia/instantsearch#5077] [Q&A] [answered]
Hi,

I have two part inside my app, **one is a place where routing is a good extra feature but the other does not need routing**. 
It is possible to disable routing per `<SearchBox>`  or how can I achieve this? 

Is the solution to create two separated `<InstanceSearch>` wrappers, one with and the other without `routing`?

Thanks 😄

**Comment by Haroenv** [comment, marked as answer]
Do those applications live on the same page, or is there another condition to why you want or not want routing? You can either have two different wrappers indeed, or conditionally disable routing.

Do you have some current code we can take a look at?

**Reply by Mat-moran** [reply]
Thanks @Haroenv for the answer, there is no code yet, we are thinking planning the app. 

The two parts are:
- SearchBox in the header of the page that shows a modal with suggestions and products to click and go direct to the product. Here I think we do not need routing, not need to save the search state here.
- Specialized search page with refinements, etc... (It would be nice to have routing. here)

I think the easiest is to wrap it inside different `<InstantSearch>` because I do not know how to disable routing base on condition. Have you got a link to documentation explaining this feature? Have you got any preferences to achieve this?

**Comment by francoischalifour** [comment]
It sounds like the experience you want to build is an [Autocomplete](https://www.algolia.com/doc/ui-libraries/autocomplete/introduction/what-is-autocomplete/) that redirects to an [InstantSearch](https://www.algolia.com/doc/guides/building-search-ui/ui-and-ux-patterns/autocomplete/react-hooks/) results page.

Autocomplete can be used as [a modal](https://www.algolia.com/doc/ui-libraries/autocomplete/core-concepts/detached-mode/), display suggestions and results, and redirect to the search results page.
